test/integration_test_helper.rb

```diff
@@ -121,3 +127,20 @@ def login_admin
     super
   end
 end
+
+class TestSSO < SSO::Base
+  delegate :session, :to => :controller
+
+  def available?
+    return true if Rails.env.test? && request.cookies['test_user'].present?
+    false
+  end
+
+  def authenticated?
+    self.user = User.current.present? ? User.current : authenticate!
+  end
+
+  def authenticate!
+    self.user = request.cookies['test_user']
```

Comment:
I think this is only called from `authenticated?` which already assign it to `self.user`